syntax = "proto3";

package externalscaler;

service ExternalScaler {
  rpc IsActive(ScaledObjectRef) returns (IsActiveResponse);
  rpc GetMetricSpec(ScaledObjectRef) returns (GetMetricSpecResponse);
  rpc GetMetrics(GetMetricsRequest) returns (GetMetricsResponse);
}

message ScaledObjectRef {
  string name = 1;
  string namespace = 2;
  map<string, string> scaler_metadata = 3;
}

message IsActiveResponse {
  bool result = 1;
}

message MetricSpec {
  string metric_name = 1;
  int64 target_size = 2;
}

message GetMetricSpecResponse {
  repeated MetricSpec metric_specs = 1;
}

message GetMetricsRequest {
  ScaledObjectRef scaled_object_ref = 1;
  string metric_name = 2;
}

message MetricValue {
  string metric_name = 1;
  int64 metric_value = 2;
}

message GetMetricsResponse {
  repeated MetricValue metric_values = 1;
}
